### PR TITLE
[DX-1220] update readme

### PR DESCRIPTION
**Pull request by Kastet**
# Summary

Restructure and simplify README.

# Why the changes

To reflect recent changes and provide better DX.

# Things worth calling out

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,161 +18,146 @@ Currently, our SDK supports interactions with our application-specific rollup ba
...
+const collection = listCollectionsResponse.data.result[0];
+
+const collectionAssetsResponse = await client.listAssets({
+  collection: collection.address,
+  pageSize: 10,
+});
 ```
 
-#### WalletConnection
+## Transactions requiring user signatures
```
Comment (zatlite) [inline README.md:77]: This heading level change shifts a lot of other headings under this heading. You might want to move other unrelated topics out from under this heading level.

**Review by ckailash — CHANGES_REQUESTED** [review]

**Review by zatlite — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -22,173 +18,177 @@ Currently, our SDK supports interactions with our application-specific rollup ba
...
 
 ```ts
-import { AlchemyProvider } from '@ethersproject/providers';
-import { Wallet } from '@ethersproject/wallet';
-import { Utils } from '@imtbl/core-sdk';
+const listCollectionsResponse = await client.listCollections({
+  pageSize: 2,
+});
+
+const collection = listCollectionsResponse.data.result[0];
```
Comment (CodeSchwert) [inline README.md:69]: Maybe we should add something about the response object should have a consistent shape (based on Axios)?

File: README.md [hunk 3]
```diff
@@ -22,173 +18,177 @@ Currently, our SDK supports interactions with our application-specific rollup ba
...
-
-  return {
-    l1Signer,
-    l2Signer,
-  };
-};
+// Create signers
+const ethWallet = new Wallet(YOUR_PRIVATE_ETH_KEY);
+const ethSigner = ethWallet.connect(provider);
+const starkSigner = Utils.createStarkSigner(YOUR_PRIVATE_STARK_KEY);
```
Comment (CodeSchwert) [inline README.md:125]: It's not clear where to get the Stark private key from? A guide or a link to Stark documentation would be helpful I think.

File: README.md (hunk 2 above)
Comment (zatlite) [reply README.md:69]: These will need to be rewritten when #172 merges

Comment (matthewmuscat) [reply README.md:69]: #172 is merged, can we get this updated?

File: README.md [hunk 4]
```diff
@@ -22,173 +18,177 @@ Currently, our SDK supports interactions with our application-specific rollup ba
...
+## Operations requiring user signatures
 
-WalletConnection is a top level connector which contains all the required resources (eg. L1 and L2 signers) to enable usage of the Core SDK workflows.
+There are two types of operations requiring user signatures:
+1. Transactions that update blockchain state
+2. Operations that Immutable X require authorization for
+
+In order to get user signatures, applications need to [generate "signers"](#how-do-applications-generate-and-use-signers).
+
+#### What are transactions that update blockchain state?
```
Comment (matthewmuscat) [inline README.md:86]: Can you add a newline here? (for code sake, not sure if it makes a difference on the formatting in preview)

File: README.md (hunk 2 above)
Comment (Kastet) [reply README.md:69]: yup I will update

File: README.md [hunk 5]
```diff
@@ -22,173 +18,177 @@ Currently, our SDK supports interactions with our application-specific rollup ba
...
 
 ```ts
 import { AlchemyProvider } from '@ethersproject/providers';
 import { Wallet } from '@ethersproject/wallet';
-import {
-  WalletConnection,
-  generateStarkWallet,
-  BaseSigner,
-} from '@imtbl/core-sdk';
+import { Utils, ImmutableX, Config } from '@imtbl/core-sdk';
```
Comment (matthewmuscat) [inline README.md]:
```suggestion
import { ImmutableX, Config } from '@imtbl/core-sdk';
```

[Utils namespace has been removed. ](https://github.com/immutable/imx-core-sdk/pull/176)

File: README.md (hunk 3 above)
Comment (matthewmuscat) [reply README.md]: Account for removed `Utils` namespace:

```suggestion
const starkSigner = createStarkSigner(YOUR_PRIVATE_STARK_KEY);
```

File: README.md [hunk 6]
```diff
@@ -22,173 +18,177 @@ Currently, our SDK supports interactions with our application-specific rollup ba
...
-      company_name,
-      contact_email,
-    },
-    iMXSignature: signature,
-    iMXTimestamp: timestamp,
-  });
-};
+const depositResponse = await client.deposit(ethSigner, {
+  type: 'ETH',
+  amount: '500000000000000000', // Amount in wei
```
Comment (matthewmuscat) [inline README.md:155]: I think we should capitalise all references of `Wei`, as it is a unit.

Comment (Kastet) [reply README.md:155]: @sprngbk

Comment (matthewmuscat) [reply README.md:155]: Discussed offline, lowercase seems to be the standard capitalization. No longer needed.

**Review by matthewmuscat — APPROVED** [review]
Looks great! Well done

**Pull request merged**